File: 프로그래머스/unrated/181930. 주사위 게임 2/주사위 게임 2.cs
using System;

public class Solution {
    public int solution(int a, int b, int c) 
    {
        int sameCount = 0;
        if(a == b) ++sameCount;
        if(b == c) ++sameCount;
        if(c == a) ++sameCount;
        
        if(sameCount == 0)
            return a + b + c;
        else if(sameCount == 1)
            return (a + b + c) * (a * a + b * b + c * c);
        else // if(sameCount == 3)
            return (a + b + c) * (a * a + b * b + c * c) *  (a * a * a + b * b * b + c * c * c);
    }
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int a, int b, int c) {
    
    int sameCount = 0;
    if (a == b) ++sameCount;
    if (b == c) ++sameCount;
    if (a == c) ++sameCount;
    
    int answer = a + b + c;
    
    if (sameCount == 0)
        return answer;
    
    answer *= (a * a + b * b + c * c);
    
    if (sameCount == 1)
        return answer;
    
    answer *= (a * a * a + b * b * b + c * c * c);
    
    return answer;
}